op-e2e/system_test.go

```diff
@@ -1570,62 +1530,137 @@ func testFees(t *testing.T, cfg SystemConfig) {
+	l1Header, err := l1.HeaderByNumber(context.Background(), nil)
 	require.Nil(t, err)
 
-	l1FeeVaultEndBalance, err := l2Seq.BalanceAt(context.Background(), predeploys.L1FeeVaultAddr, header.Number)
-	require.Nil(t, err)
+	tx, _, err := l2Seq.TransactionByHash(context.Background(), receipt.TxHash)
+	require.NoError(t, err, "Should be able to get transaction")
 
-	validatorRewardVaultEndBalance, err := l2Seq.BalanceAt(context.Background(), predeploys.ValidatorRewardVaultAddr, header.Number)
-	require.Nil(t, err)
+	var totalFee *big.Int
+	l1Fee := l1CostFn(tx.RollupCostData(), header.Time)
+	isKromaMPT := sys.RollupConfig.IsKromaMPT(header.Time)
+	if isKromaMPT {
+		// BaseFee Recipient
+		baseFeeRecipientStartBalance, err := l2Seq.BalanceAt(context.Background(), oppredeploys.BaseFeeVaultAddr, big.NewInt(rpc.EarliestBlockNumber.Int64()))
+		require.Nil(t, err)
 
-	endBalance, err := l2Seq.BalanceAt(context.Background(), fromAddr, header.Number)
-	require.Nil(t, err)
+		baseFeeRecipientEndBalance, err := l2Seq.BalanceAt(context.Background(), oppredeploys.BaseFeeVaultAddr, header.Number)
+		require.Nil(t, err)
 
-	l1Header, err := l1.HeaderByNumber(context.Background(), nil)
-	require.Nil(t, err)
+		// L1Fee Recipient
+		l1FeeRecipientStartBalance, err := l2Seq.BalanceAt(context.Background(), oppredeploys.L1FeeVaultAddr, big.NewInt(rpc.EarliestBlockNumber.Int64()))
+		require.Nil(t, err)
+
+		l1FeeRecipientEndBalance, err := l2Seq.BalanceAt(context.Background(), oppredeploys.L1FeeVaultAddr, header.Number)
+		require.Nil(t, err)
 
-	// Diff fee recipients balances
-	protocolVaultDiff := new(big.Int).Sub(protocolVaultEndBalance, protocolVaultStartBalance)
-	l1FeeVaultDiff := new(big.Int).Sub(l1FeeVaultEndBalance, l1FeeVaultStartBalance)
-	validatorRewardVaultDiff := new(big.Int).Sub(validatorRewardVaultEndBalance, validatorRewardVaultStartBalance)
+		sequencerFeeVaultStartBalance, err := l2Seq.BalanceAt(context.Background(), oppredeploys.SequencerFeeVaultAddr, big.NewInt(rpc.EarliestBlockNumber.Int64()))
+		require.Nil(t, err)
 
-	// get a validator reward scalar from KromaL1Block contract
-	l1BlockContract, err := bindings.NewKromaL1Block(predeploys.KromaL1BlockAddr, l2Seq)
-	require.Nil(t, err)
+		sequencerFeeVaultEndBalance, err := l2Seq.BalanceAt(context.Background(), oppredeploys.SequencerFeeVaultAddr, header.Number)
+		require.Nil(t, err)
+
+		// Note that coinbase address of genesis block cannot be set to SequencerFeeVault address in Kroma.
+		// So here we use header.Coinbase rather than genesis.Coinbase()
+		coinbaseStartBalance, err := l2Seq.BalanceAt(context.Background(), header.Coinbase, big.NewInt(rpc.EarliestBlockNumber.Int64()))
+		require.NoError(t, err)
 
-	validatorRewardScalar, err := l1BlockContract.ValidatorRewardScalar(&bind.CallOpts{})
-	require.Nil(t, err, "reading validatorRewardScalar")
+		coinbaseEndBalance, err := l2Seq.BalanceAt(context.Background(), header.Coinbase, header.Number)
+		require.Nil(t, err)
```

Comment:
How about just checking that the `coinbase` address in the header is the same with `sequencerFeeVault` address?